site/content/en/topics/ipam.md

```diff
@@ -0,0 +1,70 @@
+
+There are two kinds of CIDR allocator in the node IPAM controller, which are `RangeAllocator` and `CloudAllocator`.
+The `RangeAllocator` is the default one which allocates the pod CIDR for every node in the range of the cluster CIDR.
+The `CloudAllocator` allocates the pod CIDR for every node in the range of the CIDR on the corresponding VMSS or VMAS.
+
+The pod CIDR mask size of each node that belongs to a specific VMSS or VMAS is set by a specific tag 
+`{"kubernetesNodeCIDRMaskIPV4": "24"}` or `{"kubernetesNodeCIDRMaskIPV6": "64"}`. Note that the mask size tagging on 
+the VMSS or VMAS must be within the cluster CIDR, or an error would be thrown.
+
+When the above tag doesn't exist on VMSS/VMAS, the default mask size (24 for ipv4 and 64 for ipv6) would be used.
+
+To turn on the out-of-tree node IPAM controller:
+1. Disable the in-tree node IPAM controller by setting `--allocate-node-cidrs=false` in kube-controller-manager.
+1. Enable the out-of-tree counterpart by setting `--allocate-node-cidrs=true` in cloud-controller-manager.
+1. To use `RangeAllocator`:
+    * configure the `--cluster-cidr`, `--service-cluster-ip-range` and `--node-cidr-mask-size`;
+    * if you enable the ipv6 dualstack, setting `--node-cidr-mask-size-ipv4` and `--node-cidr-mask-size-ipv6` instead of 
+      `--node-cidr-mask-size`.
+1. To use `CloudAllocator`:
+    * set the `--cidr-allocator-type=CloudAllocator`;
+    * configure mask sizes of each VMSS/VMAS by tagging `{"kubernetesNodeCIDRMaskIPV4": "custom-mask-size"}` and
+      `{"kubernetesNodeCIDRMaskIPV4": "custom-mask-size"}` if necessary.
+
+## Configurations
+
+### kube-controller-manager
+
+kube-controller-manager would be configured with option `--allocate-node-cidrs=false` to disable the in-tree node IPAM controller.
+
+### cloud-controller-manager
+
+The following configurations from cloud-controller-manager would be used as default options:
+
+| name | type | default | description |
+| ----- | -----| ----- | ----- |
+| allocate-node-cidrs | bool | true | Should CIDRs for Pods be allocated and set on the cloud provider. |
+| cluster-cidr | string | "10.244.0.0/16" | CIDR Range for Pods in cluster. Requires --allocate-node-cidrs to be true. |
+| service-cluster-ip-range | string | "" | CIDR Range for Services in cluster, this would get excluded from the allocatable range. Requires --allocate-node-cidrs to be true. |
+| node-cidr-mask-size | int | 24 | Mask size for node cidr in cluster. Default is 24 for IPv4 and 64 for IPv6. |
+| node-cidr-mask-size-ipv4 | int | 24 | Mask size for IPv4 node cidr in dual-stack cluster. Default is 24. |
+| node-cidr-mask-size-ipv6 | int | 64 | Mask size for IPv6 node cidr in dual-stack cluster. Default is 64. |
+| cidr-allocator-type | string | "RangeAllocator" | The CIDR allocator type. "RangeAllocator" or "CloudAllocator". |
+
+## Limitations
+
+1. We plan to integrate out-of-tree node ipam controller with aks-engine to provider a better experience. Before that, 
```

Comment:
aks-engine doesn't support out-of-tree node ipam controller yet

Reply:
yeah, I mean I plan to integrate the out-of-tree ipam controller with aks-engine in the future.